File: server/main_server/ros_nodes/logger.cpp
// logger.cpp
#include <iostream>
#include <fstream>
#include <ctime>
#include <mutex>

std::mutex log_mutex;
std::ofstream log_file("server_log.txt", std::ios::app);

void log(const std::string& level, const std::string& message) {
    std::lock_guard<std::mutex> guard(log_mutex);

    if (!log_file.is_open()) {
        std::cerr << "Failed to open log file!" << std::endl;
        return;
    }

    // 현재 시간 구하기
    std::time_t now = std::time(nullptr);
    std::tm* local_time = std::localtime(&now);

    char time_buffer[20];
    std::strftime(time_buffer, sizeof(time_buffer), "%Y-%m-%d %H:%M:%S", local_time);

    log_file << "[" << time_buffer << "] [" << level << "] " << message << std::endl;
    log_file.flush();
}

void log_info(const std::string& message) {
    log("INFO", message);
}

void log_warning(const std::string& message) {
    log("WARNING", message);
}

void log_error(const std::string& message) {
    log("ERROR", message);
}

File: server/main_server/logger.py
# logger.py

import logging
import os

# 로그 파일 저장 위치
LOG_FILE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "server_log.txt")

# 로거 설정
logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s [%(levelname)s] %(message)s",
    handlers=[
        logging.FileHandler(LOG_FILE_PATH, encoding="utf-8"),
        logging.StreamHandler()
    ]
)

def log_info(message):
    logging.info(message)

def log_warning(message):
    logging.warning(message)

def log_error(message):
    logging.error(message)
